src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.cs

```diff
@@ -169,12 +170,20 @@ public static IReadOnlyList<Popup> GetOpenPopupsForXamlRoot(XamlRoot xamlRoot)
 
 		public static DependencyObject/* ? */ GetParent(DependencyObject reference)
 		{
+			DependencyObject realParent = null;
 #if XAMARIN
-			return (reference as _ViewGroup)?
+			realParent = (reference as _ViewGroup)?
 				.FindFirstParent<DependencyObject>();
 #else
-			return reference.GetParent() as DependencyObject;
+			realParent = reference.GetParent() as DependencyObject;
 #endif
+
+			if (realParent is null && reference is SelectorItem selectorItem)
```

Comment:
I would assume this condition is true only when we reach a `NativeListViewBase`. If I'm right, discriminating on `SelectorItem` looks weird / weak. I think here we should just support native elements and pass through them using native API to reach their managed parents.

This is really similar to https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/UIElement.Android.cs#L227, except this will also compute offsets to compute transform.

Reply:
I tried replacing the logic to use `GetVisualParent` on non-Skia and non-WASM targets, however on Skia and WASM it just gets the `DependencyObjectStore` parent, which does not work for list items, so on WASM and Skia I am keeping the existing condition currently. Maybe it then would also make sense to add similar logic into that transform computation, as it now would not work for lists on Skia and WASM either, if I am not mistaken